.github/workflows/terraform-apply.yaml

```diff
@@ -0,0 +1,57 @@
+# Based on https://github.com/canonical/sunbeam-terraform/blob/9973905d24892903fab19ce691bbc5c28c53df0a/.github/workflows/deploy.yml
+# For usage, refer to templates/terraform-deploy-usage-template.yaml
+# Applies terraform module and wait for the application to go active or blocked.
+name: Apply Terraform module
+
+on:
+  workflow_call:
+    inputs:
+      model:
+        description: The model that the charm is deployed on. Defaults to `testing`.
+        required: false
+        type: string
+        default: testing
+      channel:
+        description: The channel that the charm is deployed from. Defaults to `latest/stable`.
+        required: false
+        type: string
+        default: latest/stable
+      module-directory:
+        description: The directory that stores the Terraform module. Defaults to `./terraform`.
+        required: false
+        type: string
+        default: ./terraform
+
+jobs:
+  apply:
+    name: Apply Terraform
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: microk8s
+          channel: 1.29-strict/stable
+          microk8s-addons: "dns storage rbac metallb:10.64.140.43-10.64.140.49"
+          juju-channel: 3.4/stable
+          charmcraft-channel: latest/candidate
+      - name: Install dependencies
+        run: |
+          sudo snap install --classic terraform
+      - name: Create juju model
+        run: juju add-model ${{ inputs.model }}
+      - name: Apply terraform
+        working-directory: ${{ inputs.module-directory }}
+        run: |
+          terraform init
+          terraform apply -var "channel=${{ inputs.channel }}" -var "model_name=${{ inputs.model }}" --auto-approve
+      - name: Wait for the application to be active or blocked
+        run: |
+          juju wait-for model ${{ inputs.model }} --query='forEach(applications, app => (app.status == "active" || app.status == "blocked") )'
+      - name: Juju status
+        run: juju status
```

Comment:
Is this not redundant, since there is this [line](https://github.com/canonical/charming-actions/blob/934193396735701141a1decc3613818e412da606/dump-logs/action.yml#L15).

Reply:
I 'd keep it since the other thing runs only on failure. This way, we can always assert if needed on the status, even  on successful runs.

Reply:
You are right, I did not realize it.